openedx/frontend-app-course-authoring
# [FC-0059] UI fixes on create content

## Description

This PR adds:
- Hide the "New" button on the read-only state of a library.
- Hide the "Add component" buttons on the read-only state of a library.
- Add "Read only" badge on the read-only state of a library.

![image](https://github.com/user-attachments/assets/3bb6c946-fcd4-4b02-9f50-cf9793be40fe)


## Supporting information

- Github Issue: https://github.com/openedx/frontend-app-course-authoring/issues/1040
- Internal ticket: [FAL-3754](https://tasks.opencraft.com/browse/FAL-3754)


## Testing instructions

- Go to Studio and create a new library
- Enable public read of the library in cms bash:
```
from openedx.core.djangoapps.content_libraries.models import ContentLibrary
lib = ContentLibrary.objects.all()[0]
lib.allow_public_read = True
lib.save()
```
- Create and login with a new normal user (not staff).
- Go to the library home using the id of your new library
- Verify that the `New` button is hidden.
- Verify that all `Add component` buttons in empty states are hidden.
- Verify that you can see the "Read only" badge.
